rules/SwiftMailer/Rector/ClassMethod/Swift_MessageToEmailRector.php

```diff
@@ -0,0 +1,232 @@
+
+    /**
+     * @var array<string, string>
+     */
+    private array $basicMapping = [
+        'setSubject' => 'subject',
+        'setPriority' => 'priority',
+    ];
+
+    /**
+     * @var array<string, ?string>
+     */
+    private array $addressesMapping = [
+        'addBcc' => null,
+        'addCc' => null,
+        'addFrom' => null,
+        'addReplyTo' => null,
+        'addTo' => null,
+        'setBcc' => 'bcc',
+        'setCc' => 'cc',
+        'setFrom' => 'from',
+        'setReplyTo' => 'replyTo',
+        'setTo' => 'to',
+    ];
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Convert \Swift_Message into an Symfony\Component\Mime\Email',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+$message = (new \Swift_Message('Hello Email'))
+        ->setFrom('send@example.com')
+        ->setTo(['recipient@example.com' => 'Recipient'])
+        ->setBody(
+            $this->renderView(
+                'emails/registration.html.twig',
+                ['name' => $name]
+            ),
+            'text/html'
+        )
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+$message = (new Email())
+    ->from(new Address('send@example.com'))
+    ->to(new Address('recipient@example.com', 'Recipient'))
+    ->subject('Hello Email')
+    ->html($this->renderView(
+        'emails/registration.html.twig',
+        ['name' => $name]
+    ))
+;
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $this->traverseNodesWithCallable($node, function (Node $node) {
+            if (
+                $node instanceof ClassMethod &&
+                $node->returnType instanceof FullyQualified &&
+                $this->isName($node->returnType, self::SWIFT_MESSAGE_FQN)
+            ) {
+                $node->returnType = new FullyQualified(self::EMAIL_FQN);
+            }
+
+            if ($node instanceof Param &&
+                $node->type instanceof FullyQualified &&
+                $this->isName($node->type, self::SWIFT_MESSAGE_FQN)
+            ) {
+                $node->type = new FullyQualified(self::EMAIL_FQN);
+            }
+
+            if ($node instanceof New_) {
+                if (! $this->isName($node->class, self::SWIFT_MESSAGE_FQN)) {
+                    return null;
+                }
+                $args = $node->getArgs();
+                if (count($args) > 0) {
+                    $node = new MethodCall(new New_(new FullyQualified(self::EMAIL_FQN)), 'subject', [$args[0]]);
+                } else {
+                    $node->class = new FullyQualified(self::EMAIL_FQN);
+                }
+            }
+
+            if ($node instanceof MethodCall) {
```

Comment:
@TomasVotruba Just noticed a bug, unrelated method calls are being updated as well.

I tried adding
```php
if (!$this->isObjectType($node->var, new ObjectType(self::SWIFT_MESSAGE_FQN))) {
    return $node;
}
```
here, but that triggers a failure on daisy chained methods because `$node->var` is a MethodCall. Any suggestions?

Reply:
@stefantalen please create failing test case PR for skip fixture so we can take a look

Reply:
#604